src/NUnitEngine/nunit.engine/Services/AgentLaunchers/AgentLauncherBase.cs

```diff
@@ -0,0 +1,124 @@
+﻿// Copyright (c) Charlie Poole, Rob Prouse and Contributors. MIT License - see LICENSE.txt
+
+#if NETFRAMEWORK
+using NUnit.Engine.Extensibility;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Runtime.Versioning;
+using System.Text;
+using System.Threading.Tasks;
+using NUnit.Common;
+using TestCentric.Metadata;
+using System.IO;
+
+namespace NUnit.Engine.Services.AgentLaunchers
+{
+    public abstract class AgentLauncherBase : IAgentLauncher
+    {
+        protected abstract string AgentName { get; }
+        protected abstract TestAgentType AgentType { get; }
+        protected abstract FrameworkName TargetRuntime { get; }
```

Comment:
Can you rename this to AgentRuntime to differentiate from the TestPackage TargetFrameworkName?

Reply:
Good idea!